### DISCUSSION TITLE
Is there any way to define the subfolder for the stream?

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#5122] [FAQs]
1. When using AMS for multiple cameras streaming over weeks, the HLS files ends up creating a clutter that is hard-to-navigate on S3 or in streams folder of the application(100k+ files in a single 'folder' in S3)
2. Given that the HLS chunks don't inherently contain any timestamp and loop from 0001-->9999 and then back to 0000, the original chunks ends up being overwritten. This beats the purpose of backing up to S3, as even the backup ends up getting overwritten.

Is there any way to define subfolder and to stop overwriting of the HLS files in that subfolder?

**Comment by Mohit-3196** [comment]
Hi,
Let me try to answer the questions.

1. 

When creating a stream via REST method, a custom subfolder can be specified for each stream id. Here is a curl sample:

```
curl --location --request POST 'https://domain:5443/LiveApp/rest/v2/broadcasts/create' \
--header 'Content-Type: application/json' \
--data-raw '{ 
    "streamId":"stream1",
    "subFolder": "stream1"
}'
```
When mp4 and HLS files begin to upload, a subfolder will be created inside the streams directory in the s3 bucket.

Similarly, you can create a stream using the current date stamp subfolder, but you must delete the existing stream before creating a new one if you want to create the same stream for a different date.

```
curl --location --request POST 'https://domain:5443/LiveApp/rest/v2/broadcasts/create' \
--header 'Content-Type: application/json' \
--data-raw '{ 
    "streamId":"stream1",
    "subFolder": "stream1/14-10-2022"
}'
```

**NOTE:** If you are using the subfolder to create a custom path, then the HLS or MP4 URL would be as follows using the embedded player (play.html).

```
http://domain:5080/LiveApp/play.html?name=subFolderName/test&playOrder=hls

http://domain:5080/LiveApp/play.html?name=subFolderName/test&playOrder=vod
```

2. We have discussed the second part of this question here: https://github.com/orgs/ant-media/discussions/4875#discussioncomment-5231880

I hope it helps.

Thank you
